backend/controllers/addAppointmentDetails.js

```diff
@@ -23,6 +23,67 @@ const setSelected = (value, items) =>
         : item
   )
 
+const buildDateTime = ({ date, hours, minutes }) => {
+  const time = date && hours && minutes && moment(date, DAY_MONTH_YEAR)
+  return time && time.hour(Number(hours)).minutes(Number(minutes))
+}
+
+const extractHoursMinutes = dateTime => {
+  const instant = moment(dateTime, DATE_TIME_FORMAT_SPEC)
+  const hours = instant.hours()
+  const minutes = instant.minutes()
+
+  return {
+    hours: hours <= 9 ? `0${hours}` : hours,
```

Comment:
this looks complicated and can return either a string or number (string if < 10).  You've got an `instant` anyway so think it would be better to do:
```javascript
hours: instant.format('HH')
minutes: instant.format('mm')
```